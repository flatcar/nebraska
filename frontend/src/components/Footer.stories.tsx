import { createStore } from '@reduxjs/toolkit';
import { Meta, StoryFn } from '@storybook/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import FooterComponent from './Footer';

export default {
  title: 'Footer',
  component: FooterComponent,
  argTypes: {},
  decorators: [
    Story => {
      return (
        <MemoryRouter>
          <Story />
        </MemoryRouter>
      );
    },
  ],
} as Meta;

const Template: StoryFn = args => {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const store = createStore((state = { config: {} }, _action) => state, {
    config: {
      ...args,
    },
  });
  return (
    <Provider store={store}>
      <FooterComponent />
    </Provider>
  );
};

export const FooterNoOverride = {
  render: Template,

  args: {
    title: '',
    nebraska_version: '',
  },
};

export const FooterOverride = {
  render: Template,

  args: {
    title: 'Some Pro Update Service',
    nebraska_version: '1.2.3',
  },
};
