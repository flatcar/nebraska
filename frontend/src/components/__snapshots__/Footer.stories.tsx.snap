// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer FooterNoOverride smoke-test 1`] = `
<div class="MuiBox-root css-sw8jpj">
  Nebraska
</div>
`;

exports[`Footer FooterOverride smoke-test 1`] = `
<div class="MuiBox-root css-sw8jpj">
  Some Pro Update Service 1.2.3
</div>
`;
